README.md

```diff
@@ -0,0 +1,124 @@
+# rhea-promise
+
+A Promisified layer over rhea AMQP client.
+
+### Examples
+
+#### Sending a message.
+```ts
+import {
+  Connection, Sender, EventContext, Message, ConnectionOptions, Delivery, SenderOptions
+} from "rhea-promise";
+
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const host = process.env.AMQP_HOST || "host";
+const username = process.env.AMQP_USERNAME || "username";
+const port = parseInt(process.env.AMQP_PORT || "5671");
+const senderAddress = process.env.SENDER_ADDRESS || "address";
+async function main(): Promise<void> {
+  const connectionOptions: ConnectionOptions = {
+    transport: "tls",
+    host: host,
+    hostname: host,
+    username: username,
+    port: port,
+    reconnect: false
+  };
+  const connection: Connection = new Connection(connectionOptions);
+  const senderName = "sender-1";
+  const senderOptions: SenderOptions = {
+    name: senderName,
+    target: {
+      address: senderAddress
+    },
+    onError: (context: EventContext) => {
+      const senderError = context.sender && context.sender.error;
+      if (senderError) {
+        console.log("[%s] An error occurred for sender '%s': %O.",
+          connection.id, senderName, senderError);
+      }
+    },
+    onSessionError: (context: EventContext) => {
```

Comment:
why is session error specifically broken out, and if so, why not also errors on the connection itself?